.travis.yml

```diff
@@ -74,6 +98,69 @@ script: |
       git diff
       rebuild cfitsio
       ;;
+    versiondb*)
+      echo -e "*** testing VERSIONDB_PUSH=$VERSIONDB_PUSH for versiondb ***\n"
+      export VERSIONDB_PUSH
+      export VERSIONDB_REPO=./versiondb-test
+      mkdir -p $VERSIONDB_REPO
+      (cd $VERSIONDB_REPO; git init --bare)
+      # pull will fail unless there is a master ref to actually pull; so we
+      # make one
+      (
+        tmpdir=$(mktemp -u -t 'versiondb.XXXXXXXX')
+        trap "{ rm -rf $tmpdir; }" EXIT
+
+        mkdir -p "$tmpdir"
+        git clone "$VERSIONDB_REPO" "$tmpdir"
+        cd $tmpdir
+        # it appears that with some versions of git, --author doesn't bypass
+        # the user.email/user.name check.
+        git config --local user.email 'author@example.com'
+        git config --local user.name 'A U Thor'
+        git commit --allow-empty -m 'initial commit' --author='A U Thor <author@example.com>'
+        git push origin master
+      )
+
+      ./bin/deploy
+
+      (cd versiondb; mkdir dep_db ver_db manifests)
+
+      . ./bin/setup.sh
+      rebuild cfitsio
+
+      cd $VERSIONDB_REPO
+      if [[ $VERSIONDB_PUSH == true ]]; then
+        echo 'should have master ref'
+        git show-ref --heads | grep -q refs/heads/master
+
+        revs=($(git rev-list master))
+
+        # the first commit is a garage commit in order to create the master ref
```

Comment:
The `git commit --allow-empty -m 'initial commit'` commit so that `git pull` doesn't blow up because there aren't any refs in the remote to fetch.  Do you have a suggestion on how to improve the comment?